### DISCUSSION TITLE
Trouble making autonuke .iso for booting on out-of-band managment

### DISCUSSION THREAD

**Discussion by Steeze-Fry** [PartialVolume/shredos.x86_64#141] [Q&A]
Great work on everything, I'm really liking ShredOS!

My goal is to make an .iso that will autonuke. To be mounted and booted on servers with out-of-band management (iDRAC) Things I can't physically put a USB or CD into.
I'm not very familiar with compiling and making new bootable .iso so I'm not sure if I'm going about things totally wrong or I'm missing something simple.

I did managed to setup a flash drive to successfully autonuke.
So at least I know I have the syntax right. 

My first thought was to match the commands that worked on the usb. Just open the .iso (in PowerISO) and edit the grub.cfg at: 
/boot/grub/grub.cfg 
and 
/EFI/BOOT/grub.cfgto 
but /EFI/BOOT/grub.cfgto was not there.
So I just edited the one .cfg with the same settings I did on the USB. 
When I tried to test the .iso file in a Hyper-V vm, it did not boot at all.

Tried several other ideas but still haven't figured it out.

Thanks :)
Sorry if I missed the answer somewhere else.

**Comment by PartialVolume** [comment]
@jack-kondreck 
>When I tried to test the .iso file in a Hyper-V vm, it did not boot at all.

That might be because ShredOS needs to have the following enabled.

>Symbol: FB_HYPERV [=n]                                                                                                                      │  
  │ Type  : tristate                                                                                                                            │  
  │ Defined at drivers/video/fbdev/Kconfig:2182                                                                                                 │  
  │   Prompt: Microsoft Hyper-V Synthetic Video support                                                                                         │  
  │   Depends on: HAS_IOMEM [=y] && FB [=y] && HYPERV [=n]                                                                                      │  
  │   Location:                                                                                                                                 │  
  │     -> Device Drivers                                                                                                                       │  
  │       -> Graphics support                                                                                                                   │  
  │         -> Frame buffer Devices                                                                                                             │  
  │ (1)       -> Support for frame buffer devices (FB [=y])                                                                                     │  
  │ Selects: FB_CFB_FILLRECT [=y] && FB_CFB_COPYAREA [=y] && FB_CFB_IMAGEBLIT [=y] && FB_DEFERRED_IO [=y] && DMA_CMA [=y]

and

>Symbol: HYPERV [=n]                                                                                                                         │  
  │ Type  : tristate                                                                                                                            │  
  │ Defined at drivers/hv/Kconfig:5                                                                                                             │  
  │   Prompt: Microsoft Hyper-V client drivers                                                                                                  │  
  │   Depends on: X86 [=y] && ACPI [=y] && X86_LOCAL_APIC [=y] && HYPERVISOR_GUEST [=y]                                                         │  
  │   Location:                                                                                                                                 │  
  │     -> Device Drivers                                                                                                                       │  
  │ (3)   -> Microsoft Hyper-V guest support                                                                                                    │  
  │ Selects: PARAVIRT [=y] && X86_HV_CALLBACK_VECTOR [=y]

Does you modified .iso if burned to CD/DVD boot normally on a legacy system?

Depending upon whether you are running Linux and are comfortable building from source this is how I would do it.

If I was building a autonuke .iso I would probably do it by starting with the existing ShredOS software and then following the instructions below make the changes so it builds an .iso instead of a .img. ( I really need to put some detailed info into the README.md  that explains what you need to do to build an .iso )

Without explaining in detail though, these are the basics.

#### Building a legacy boot .iso for cd/dvd (not UEFI) 
```
Add the autonuke option  to ../board/shredos/grub.cfg

Using make menuconfig..
1. IMPORTANT ! Check the directory in build options, change to the appropriate location of the buildroot you are actually working on. This is relevant for me when I'm copying the .img build to the .iso build in a different directory. 
2. Select Build Options, then change the directory from /media/jbloggs/Software/shredos/Shredos.2020.05/shredos.2020.02/configs/shredos_defconf  to /media/jbloggs/Software/shredos/Shredos.2020.05.iso/shredos.2020.02/configs/shredos_defconf , i.e. the only difference is appending .iso so the path becomes  ../Shredos.2020.05.iso/..
3. Select bootloaders->Platform-> i386-pc (old bios boot method)
4. Change Boot Partition from 'hd0,msdos1' to 'cd'
5. Select bootloaders->builtin modules include biosdisk & iso9660
6. Select Filesystem Image->iso image-> grub2
7. Save and then exit make menuconfig.
8. make savedefconfig
9. cp board/shredos/grub.cfg  fs/iso9660/grub.cfg
10. Edit fs/iso9660/grub.cfg and replace /boot/shredos with /boot/bzImage (IMPORTANT) Get the case correct, that's a capital 'I'  not 'i' in bzImage.
11. Make sure the 32 bit futex fix is added on 32 bit architecture. CONFIG_COMPAT_32BIT_TIME see keep note.  To determine whether its installed run make linux-menuconfig and search using '/' for CONFIG_COMPAT_32BIT_TIME , if its not set as 'Y' then goto (-> General architecture-dependent options) and select it.
12. make savedefconfig
13. make clean
14. make shredos_defconfig
15. make
16. Once completed and with no errors. cd output/images
17.  cp rootfs.iso9660 to the name of the .img file but instead of .img use the .iso extension.
```

**Reply by Steeze-Fry** [reply]
After several attempts I finally figured it out thank you for all your help!!!
